LocalFolder/explainer.md

```diff
@@ -0,0 +1,129 @@
+* **Current version: this document**
+
+## Introduction
+[UWP apps](https://learn.microsoft.com/en-us/windows/uwp/get-started/universal-application-platform-guide) often make use of the [`ApplicationData.LocalFolder`](https://learn.microsoft.com/en-us/uwp/api/windows.storage.applicationdata.localfolder) API for access to local storage. This API manages storage in a system file directory unique to the Package Family Name (PFN) of the installed app package. A UWP app distributed by the Microsoft Store may be replaced with a PWA app under the same PFN. When this happens, existing files in the `LocalFolder` directory become inaccessible to the PWA. 
+
+In this explainer, we propose a solution that allows Microsoft Store PWAs to read and delete files from the `LocalFolder` directory belonging to their PFN. This allows apps to provide a more seamless user experience after an update as well as reclaim storage space.
+
+The proposed solution works by making use of the [Origin Private File System (OPFS)](https://web.dev/articles/origin-private-file-system) and the [File System Access API](https://wicg.github.io/file-system-access/). It exposes an app's LocalFolder file system directory as an entry in the app origin's OPFS root directory.
+
+## Goals
+
+* Enable a Microsoft Store PWA to:
+   * read files from the `LocalFolder` directory specific to its PFN.
+   * delete any file and sub-directory in its `LocalFolder` directory. 
+* Avoid prompting end users to provide input or confirmation.
+* Avoid naming collisions with other OPFS root entries. 
+
+## Non-goals
+
+* Enable a Microsoft Store PWA to create new files or modify existing files in its LocalFolder directory.
+* Support web apps not installed and managed by the Microsoft Store.
+* Support web apps on non-Windows platforms.
+* Standardization as a web specification.
+* Introduce a non-standard Web API
+
+## Proposed Solution 
+
+### Existing behavior
+
+`navigator.storage.getDirectory()` is an existing API that returns a promise to a FileSystemDirectoryHandle representing the root of the origin's OPFS storage space. By default, an origin's OPFS root directory will have no entries.
+
+### New behavior
+
+Within a Microsoft Store PWA context, the directory handle for OPFS root will contain an additional entry that represents the app's LocalFolder location. This entry can be retrieved as a `FileSystemDirectoryHandle` by calling `.entries()` or `.getDirectory(...)` on the OPFS root directory handle.
+
+The LocalFolder directory handle and its contents will have properties different from other OPFS directory handles - they will only have read and delete permissions instead of read and write permissions.
+
+```JS
+    // .getDirectory()
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        let localStorage = await opfsRoot.getDirectoryHandle("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", {create: false}); 
```

Comment:
You're explicitly calling out the optional parameter of `{create: false}` here.  What would happen if you passed in `{create: true}'?  We should document that (or at least have a comment in this code snippet to address that).

Reply:
Added note.